internal/federation/handle.go

```diff
@@ -38,37 +38,52 @@ func MakeJoinRequestsHandler(s *Server, w http.ResponseWriter, req *http.Request
 		return
 	}
 
+	makeJoinResp, err := MakeRespMakeJoin(s, room, userID)
+	if err != "" {
+		w.WriteHeader(500)
+		w.Write([]byte(err))
+		return
+	}
+
+	// Send it
+	res := map[string]interface{}{
+		"event":        makeJoinResp.JoinEvent,
+		"room_version": makeJoinResp.RoomVersion,
+	}
```

Comment:
pretty sure you can just `json.Marshall(makeJoinResp)` rather than copying to this intermediate map

Reply:
Thank you, that worked.